extension.js

```diff
@@ -674,7 +830,11 @@ class PidginClient extends GObject.Object {
 
 	enableSearchProvider() {
 		if (this._searchProvider == null) {
-			this._searchProvider = new PidginSearchProvider(this);
+			let [major, minor] = Config.PACKAGE_VERSION.split('.').map(s => Number(s));
+			if (major >= 43)
+				this._searchProvider = new PidginSearchProvider(this);
+			else
+				this._searchProvider = new PidginLegacySearchProvider(this);
```

Comment:
I wonder if we really need to support both providers. We can raise extension version and declare that the minimum supported shell version is 43. This way we can completely remove `PidginLegacySearchProvider`.

WDYT?

Reply:
As you want, but I'm a Debian stable user and I've often gotten a few late versions of Gnome. So I know well the frustrating situation of seeing the latest versions of my favorite extensions no longer supported the version that my Debian imposes on me. That's why I tried to maintain compatibility with Gnome versions < 43! :wink:

__Note:__ In facts, in Debian Stable, we currently have Gnome 43.4.

Reply:
Also, isn't the `Config.PACKAGE_VERSION` is the extension version and not the shell version? Looks like extension version is not really helpful here - we need to know shell version to choose search provider implementation.

Reply:
This variable name is actually misleading, but it's the [official way](https://gjs.guide/extensions/upgrading/gnome-shell-40.html#checking-gnome-shell-version) to check Gnome Shell version.

Reply:
Cool. And one last thing, please increase the extension version.

Reply:
[done](https://github.com/muffinmad/pidgin-im-gnome-shell-extension/pull/61/commits/7fca195623c4dab552e1e8cfb4dbf343d6476072) !